### PR TITLE
[Pet Set] Fix losing pet updated information between update retries

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

Address #32481

@bprashanth 

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32482)

### CODE REVIEW

File: pkg/controller/petset/pet.go [hunk 1]
```diff
@@ -197,24 +197,28 @@ func (p *apiServerPetClient) Create(pet *pcb) error {
...
-	pod, needsUpdate, err := copyPetID(pet, expectedPet)
-	if err != nil || !needsUpdate {
-		return err
-	}
-	glog.Infof("Resetting pet %v to match PetSet %v spec", pod.Name, pet.parent.Name)
-	for i, p := 0, &pod; ; i++ {
-		_, updateErr = pc.Update(p)
-		if updateErr == nil || i >= updateRetries {
-			return updateErr
+	glog.Infof("Resetting pet %v/%v to match PetSet %v spec", pet.pod.Namespace, pet.pod.Name, pet.parent.Name)
```
Comment (bprashanth) [inline pkg/controller/petset/pet.go]: please move this line to after the !needsUpdate return so we don't spam the logs when we're not actually doing anything

File: pkg/controller/petset/pet.go [hunk 2]
```diff
@@ -197,24 +197,28 @@ func (p *apiServerPetClient) Create(pet *pcb) error {
...
+	for rep := 0; rep < updateRetries; rep++ {
```
Comment (bprashanth) [inline pkg/controller/petset/pet.go]: nit: please rename loop variable to `i`

File: pkg/controller/petset/pet.go [hunk 3]
```diff
@@ -197,24 +197,28 @@ func (p *apiServerPetClient) Create(pet *pcb) error {
...
+		updatePod, needsUpdate, err := copyPetID(pet, expectedPet)
+		if err != nil || !needsUpdate {
+			return err
+		}
+		_, updateErr = pc.Update(&updatePod)
+		if updateErr == nil {
```
Comment (bprashanth) [inline pkg/controller/petset/pet.go]: please more the rep < updateRetries exit condition to here like it was before, you're wasting a Get now in the case where we exit the loop

**Comment by bprashanth** [comment]
@m1093782566 Thanks. if you have time please add a unittests the ensures we fix identity if something corrupts it

File: pkg/controller/petset/pet.go (hunk 1 above)
Comment (m1093782566) [inline pkg/controller/petset/pet.go]: Okay, fixed.

File: pkg/controller/petset/pet.go (hunk 2 above)
Comment (m1093782566) [inline pkg/controller/petset/pet.go]: Okay, fixed.

File: pkg/controller/petset/pet.go (hunk 3 above)
Comment (m1093782566) [inline pkg/controller/petset/pet.go]: Okay, fixed.

**Comment by m1093782566** [comment]
comments addressed. PTAL.

> if you have time please add a unittests the ensures we fix identity if something corrupts it

I find the current test cases does cover the real `func (p *apiServerPetClient) Update(pet *pcb, expectedPet *pcb) (updateErr error)` in pet.go, it use `func (f *fakePetClient) Update(expected, wanted *pcb) error` in fakes.go instead.

So, add unit test cases for real `Update()` is a huge task and may out of scope in this PR.

I am glad to implement test cases for petset in another PR.

**Comment by m1093782566** [comment]
ping @bprashanth for another review :)

Thanks!

**Comment by bprashanth** [comment]
LGTM thanks

**Comment by m1093782566** [comment]
Test flakes:

k8s.io/kubernetes/pkg/registry/replicaset/etcd TestWatch 37s

go test -v k8s.io/kubernetes/pkg/registry/replicaset/etcd -run TestWatch$

resttest.go:1251: unexpected timeout from result channelfrom

**Comment by m1093782566** [comment]
@bprashanth slightly modify one line to pass go fmt check.

BTW, CI failed due to flake case #32770. Would you please ping robot?

Thanks!

**Pull request merged**